test/react/functional-components/refs.js

```diff
@@ -0,0 +1,30 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+function Child(props) {
+  return <div ref={props.forwardedRef} />;
+}
+
+const WrappedComponent = React.forwardRef((props, ref) => {
+  return <Child {...props} forwardedRef={ref} />;
+});
+
+function App() {
+  var x = React.createRef();
+
+  return (
+    <WrappedComponent ref={x} />
+  );
+}
+
+App.getTrials = function(renderer, Root) {
+  renderer.update(<Root />);
+  return [['simple render with refs', renderer.toJSON()]];
```

Comment:
It's not obvious from this test that the ref forwarding actually works. Can you make it do something with that ref? e.g. pass it down a few levels to a class, call a method to read a prop.

Reply:
I originally tried, but the `test-renderer` doesn't seem to work with object refs right now so I was going to look into that after this landed.

Reply:
I'll add a callback ref to show it working though.